scripts/lab/run_recall_test.py

```
@@ -158,31 +158,64 @@ def restore_config(original: Dict[str, Optional[str]]) -> None:
-                pass
-            time.sleep(1)
+
+    # Wait for API to be healthy (fail loudly on timeout)
+    last_exc = None
+    for attempt in range(60):
+        try:
+            resp = requests.get(f"{API_URL}/health", timeout=2)
+            if resp.status_code == 200:
+                print(f"API ready after config change ({attempt}s)")
+                return
+        except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)